src/main/java/com/iwaconsolti/league/manager/controller/LeaguesController.java

```diff
@@ -0,0 +1,151 @@
+package com.iwaconsolti.league.manager.controller;
+
+import com.iwaconsolti.league.manager.model.Matches;
+import com.iwaconsolti.league.manager.model.Players;
+import com.iwaconsolti.league.manager.model.Teams;
+import com.iwaconsolti.league.manager.service.ILeagues;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/leagues/{leagueType}")
+@Slf4j
+public class LeaguesController {
+
+    private ILeagues soccerLeagues;
+    private ILeagues baseBallLeague;
+
+    public LeaguesController(@Qualifier("soccerLeague") ILeagues soccerLeague, @Qualifier("baseBallLeague") ILeagues baseBallLeague) {
+        this.soccerLeagues = soccerLeague;
+        this.baseBallLeague = baseBallLeague;
+        log.info("Default LeaguesController created");
+    }
+
+    @PostMapping("/team")
+    public ResponseEntity<Teams> createTeam(@PathVariable String leagueType, @RequestBody Teams newTeam) {
+        if (leagueType.equals("soccer")) {
+            return ResponseEntity.ok(soccerLeagues.saveTeams(newTeam));
```

Comment:
In the POST methods, the ResponseEntity could be changed to a 201 message to indicate that the resource was created.

```suggestion
           if (leagueType.equals("soccer")) {
            return ResponseEntity.status(HttpStatus.CREATED).body(soccerLeagues.saveTeams(newTeam));
        } else if (leagueType.equals("baseball")) {
            return ResponseEntity.status(HttpStatus.CREATED).body(baseBallLeague.saveTeams(newTeam));
        } else {
            log.warn("League not found");
            return ResponseEntity.notFound().build();
        }
```

In addition, it could be considered sending an error message when there are already 10 computers, it could be a 413 Payload Too Large or 409 Conflict message.